roles/kubernetes-apps/policy_controller/meta/main.yml

```diff
@@ -4,13 +4,13 @@ dependencies:
     when:
       - kube_network_plugin == 'calico'
       - enable_network_policy
-      - calico_datastore != "kdd"
+      - calico_datastore != "kdd" or calico_policy_version is version('v3.6.0', '>=')
     tags:
       - policy-controller
 
   - role: policy_controller/calico
     when:
       - kube_network_plugin == 'canal'
-      - calico_datastore != "kdd"
+      - calico_datastore != "kdd" or calico_policy_version is version('v3.6.0', '>=')
```

Comment:
Ditto

Reply:
Shouldn't we just leave one dependency on this role for both calico and canal with `enable_network_policy` set to  true? It makes little sense to me otherwise. Something like:
```
---
dependencies:
  - role: policy_controller/calico
    when:
      - kube_network_plugin in ['calico', 'canal']
      - enable_network_policy
    tags:
      - policy-controller
```

Reply:
@wayfrro haha exactly what I wrote yesterday too https://github.com/kubernetes-sigs/kubespray/pull/9358#issuecomment-1268340581

Reply:
Indeed it's cleaner that way but the original intent of the PR you are referencing was a clean up which accidentally triggered this bug.

@wayfrro please update the PR with the code you proposed above which better communicates the intention of having the policy controller dependency on both calico and canal.